templates/products/products-listing-row.php

```diff
@@ -126,26 +126,25 @@
     </td>
     <td class="post-date" data-title="<?php esc_attr_e( 'Date', 'dokan-lite' ); ?>">
         <?php
+
+
+
         if ( '0000-00-00 00:00:00' == $post->post_date ) {
-            $t_time    = $h_time    = __( 'Unpublished', 'dokan-lite' );
+            $post_published_date = $human_readable_time = __( 'Unpublished', 'dokan-lite' );
             $time_diff = 0;
         } else {
```

Comment:
a better approach without all the complexities:
```
// get current time
    $current_time = dokan_current_datetime();
    // set timezone to gmt
    $gmt_time = $current_time->setTimezone( new DateTimeZone( 'UTC' ) );
    // read post gmt time
    $post_time_gmt = $gmt_time->modify( $post->post_date_gmt );

    // get post time in wp timezone
    $post_published_ts = $post_time_gmt->setTimezone( dokan_wp_timezone() )->getTimestamp();
    $format = apply_filters( 'dokan_date_time_format', wc_date_format() . ' ' . wc_time_format() );
    // currently dokan_format_date doesn't support time format, will update this soon, 
    // right now we need to send $format from our end.
    $post_published_date = dokan_format_date( $post_published_ts, $format );

    // get human readable time
    $time_diff = $current_time->getTimestamp() - $post_time_gmt->getTimestamp();
    if ( $time_diff > 0 && $time_diff < 24 * 60 * 60 ) {
        $human_readable_time = sprintf( __( '%s ago', 'dokan-lite' ), human_time_diff( $post_time_gmt->getTimestamp() ) );
    }
```

Reply:
Refactored the code block with your code above.

Reply:
@shohag121 thanks, @FaraziF bhai can you please test this or one more time ?